synapse/lib/node.py

```diff
@@ -46,15 +46,29 @@ def __init__(self, snap, buid=None, rawprops=None):
     def __repr__(self):
         return f'Node{{{self.pack()}}}'
 
-    async def storm(self, text, opts=None, user=None):
+    async def storm(self, text, opts=None, user=None, path=None):
+        '''
+        Args:
+            path (Path):
+                If set, then vars from path are copied into the new runtime, and vars are copied back out into path
+                at the end
+
+        Note:
+            If opts is not None and opts['vars'] is set and path is not None, then values of path vars take precedent
+        '''
         query = self.snap.core.getStormQuery(text)
-        with self.snap.getStormRuntime(opts=opts, user=user) as runt:
+        newopts = dict(opts) if opts else {}
+        if path:
+            newopts['vars'] = {**newopts.get('vars', {}), **path.vars}
```

Comment:
Any speed benefit of the dict comprehension is probably scrubbed here because it forces a new dictionary construction even when newopts *has* a vars dict.  ( the dangers of  foo.get('baz', {}) ) but we can let it ride :)

Reply:
I was concerned that I needed a fresh dict anyway because I didn't want the source modified. However, I see that the Runtime constructor does copy out the values. Before, I had a "free" copy, but since I don't need to copy, the fancy pants stuff isn't necessary. Will change.